target/linux/realtek/dts-5.10/rtl9302_zyxel_xgs1210-12.dts

```diff
@@ -0,0 +1,326 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/dts-v1/;
+
+#include "rtl930x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "zyxel,xgs1210-12", "realtek,rtl838x-soc";
+	model = "Zyxel XGS1210-12 Switch";
+
+	// mw.l 0xb8003308 0x00200000 active low toggles PHY reset
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	i2c0: i2c-rtl9300 {
+		compatible = "realtek,rtl9300-i2c";
+		reg = <0x1b00036c 0x3c>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		sda-pin = <9>;
+		scl-pin = <8>;
+	};
+
+	i2cmux {
+		compatible = "realtek,i2c-mux-rtl9300";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		i2c-parent = <&i2c0>;
+
+		/* i2c of the left SFP cage: port 11 */
+		i2c01: i2c-rtl9300-1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <9>;
+			scl-pin = <8>;
+		};
+
+		/* i2c of the right SFP cage: port 12 */
+		i2c02: i2c-rtl9300-2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			sda-pin = <10>;
+			scl-pin = <8>;
+		};
+	};
+
+	sfp0: sfp-p11 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c01>;
+		los-gpio = <&gpio0 13 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio0 11 GPIO_ACTIVE_HIGH>;
+	};
+
+	sfp1: sfp-p12 {
+		compatible = "sff,sfp";
+		i2c-bus = <&i2c02>;
+		los-gpio = <&gpio0 17 GPIO_ACTIVE_HIGH>;
+		mod-def0-gpio = <&gpio0 16 GPIO_ACTIVE_LOW>;
+		tx-disable-gpio = <&gpio0 15 GPIO_ACTIVE_HIGH>;
+	};
+
+	// Reset-button: GPIO 22
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		mode {
+			label = "reset";
+			gpios = <&gpio0 22 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	led_set: led_set@0 {
+		compatible = "realtek,rtl9300-leds";
+		led_set0 = <0x0000 0xffff 0x0a20 0x0b80>; // LED set 0: 1000Mbps,  10/100Mbps
+		led_set1 = <0x0a80 0x0a20 0x0a08 0x0a08>; // LED set 1: 100MBit, 1000MBit, 2.5G
+		led_set2 = <0x0000 0xffff 0x0a20 0x0a01>; // LED set 2: 1000MBit, 10GBit
+	};
+};
+
+&spi0 {
+	status = "okay";
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
What is this frequency based off on? According to the datasheet, our chip has a max freq of 133MHz, so usually this property indicates this.

Reply:
No, this is limited by the SPI frequency of the SoC. The datasheet actually says it is fixed to 50MHz for the RTL9300 and the RTL838x. On the other hand, the MAX3421 can only run with a maximum frequency of the SPI bus of 26MHz, and that is what I see e.g. on the Netgear GS728TPv2, where I was able to measure. So I believe the actual clock frequency is probably only 25MHz. The good news is that the SPI driver does not even try to set a different frequency divider than u-boot (which in the SDK is a relatively high number, definitely not 1) and there is no way to change it via the driver API. So this property is entirely bogus. But I will change it to 50MHz as suggested by the datasheet as the maximum SPI frequency for both SoC and Flash.